opengever/maintenance/scripts/active_user_stats.py

```diff
@@ -0,0 +1,101 @@
+"""
+Script that calculates the current active users in total over all admin units an writes it into
+a logfile.
+
+bin/instance0 run src/opengever.maintenance/opengever/maintenance/scripts/active_user_stats.py
+"""
+from datetime import datetime
+from ftw.contentstats.logger import get_log_dir_path
+from logging import FileHandler
+from opengever.maintenance import dm
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.meeting import is_meeting_feature_enabled
+from opengever.meeting.model import Committee
+from opengever.ogds.models.group import Group
+from opengever.ogds.models.group import groups_users
+from opengever.ogds.models.org_unit import OrgUnit
+from opengever.ogds.models.user import User
+from opengever.workspace import is_workspace_feature_enabled
+from os.path import join
+from plone import api
+from tzlocal import get_localzone
+import json
+import logging
+import pytz
+
+logger = logging.getLogger('opengever.maintenance')
+LOG_TZ = get_localzone()
+
+
+class UserStatsCalculator(object):
+    """Returns statistics for the active users on the system
+    """
+    def get_stats(self):
+        stats = {
+            'total_active_unique_gever_users': 0,
+            'total_active_unique_teamraum_users': 0,
+            'total_active_unique_spv_users': 0,
+        }
+
+        if not is_workspace_feature_enabled():
+            stats['total_active_unique_gever_users'] = self.calc_total_active_unique_gever_users()
+        else:
+            stats['total_active_unique_teamraum_users'] = self.calc_total_active_unique_teamraum_users()
+
+        if is_meeting_feature_enabled():
+            stats['total_active_unique_spv_users'] = self.calc_total_active_unique_spv_users()
+
+        return stats
+
+    def get_active_users_by_groups_query(self, group_ids):
+        query = User.query.join(groups_users).join(Group)
+        return query.filter(Group.groupid.in_(group_ids)).filter_by(active=True)
+
+    def calc_total_active_unique_gever_users(self):
+        group_ids = [
+            org_unit.users_group_id for org_unit in
+            OrgUnit.query.filter_by(enabled=True).all()]
+
+        query = self.get_active_users_by_groups_query(group_ids)
+        return query.count()
+
+    def calc_total_active_unique_spv_users(self):
+        group_ids = [
+            committee.group_id for committee in
+            Committee.query.filter_by(workflow_state='active')]
+
+        query = self.get_active_users_by_groups_query(group_ids)
+        return query.count()
+
+    def calc_total_active_unique_teamraum_users(self):
+        role_manager = api.portal.get_tool('acl_users').portal_role_manager
+        group_ids = [principal[0] for principal in role_manager.listAssignedPrincipals('WorkspacesUser')]
```

Comment:
Why are you getting the group ids from the role assignments here and not using the same approach as for GEVER users where you get the group ids from the org unit's `users_group_id`?

Reply:
You're right. I thought that that's the way for teamraum.

I updated the script 👍